camel/storages/vectordb_storages/qdrant.py

```diff
@@ -0,0 +1,308 @@
+    ) -> None:
+        r"""Adds a list of vectors to the specified collection.
+
+        Args:
+            vectors (List[VectorRecord]): List of vectors to be added.
+            **kwargs (Any): Additional keyword arguments.
+
+        Raises:
+            RuntimeError: If there was an error in the addition process.
+        """
+        self.validate_vector_dimensions(records)
+        qdrant_points = [PointStruct(**asdict(p)) for p in records]
+        op_info = self._client.upsert(collection_name=self.collection,
+                                      points=qdrant_points, wait=True,
+                                      **kwargs)
+        if op_info.status != UpdateStatus.COMPLETED:
+            raise RuntimeError(
+                "Failed to add vectors in Qdrant, operation info: "
+                f"{op_info}")
+
+    def delete(
+        self,
+        ids: List[str],
+        **kwargs: Any,
+    ) -> None:
+        r"""Deletes a list of vectors identified by their IDs from the storage.
+
+        Args:
+            ids (List[str]): List of unique identifiers for the vectors to be
+                deleted.
+            **kwargs (Any): Additional keyword arguments.
+
+        Raises:
+            RuntimeError: If there is an error during the deletion process.
+        """
+        points = cast(List[Union[str, int]], ids)
+        op_info = self._client.delete(
+            collection_name=self.collection,
+            points_selector=PointIdsList(points=points),
+            wait=True,
+            **kwargs,
+        )
+        if op_info.status != UpdateStatus.COMPLETED:
+            raise RuntimeError(
+                "Failed to delete vectors in Qdrant, operation info: "
+                f"{op_info}")
+
+    def query(
+        self,
+        query: VectorDBQuery,
+        **kwargs: Any,
+    ) -> List[VectorDBQueryResult]:
+        r"""Searches for similar vectors in the storage based on the provided query.
+
+        Args:
+            query (VectorDBQuery): The query object containing the search
+                vector and the number of top similar vectors to retrieve.
+            **kwargs (Any): Additional keyword arguments.
+
+        Returns:
+            List[VectorDBQueryResult]: A list of vectors retrieved from the
+                storage based on similarity to the query vector.
+        """
+        # TODO: filter
+        search_result = self._client.search(
+            collection_name=self.collection,
+            query_vector=query.query_vector,
+            with_payload=True,
+            with_vectors=True,
+            limit=query.top_k,
+            **kwargs,
+        )
+        query_results = []
+        for point in search_result:
+            query_results.append(
+                VectorDBQueryResult.construct(
+                    similarity=point.score,
+                    id=str(point.id),
+                    payload=point.payload,
+                    vector=point.vector,  # type: ignore
```

Comment:
Just one simple question, in the definition of VectorDBQueryResult.construct, we need a "class", but here it seems you don't specify the class, that's why you use `# type: ignore` here?

Reply:
If I understand your question correctly, `VectorDBQueryResult.construct` actually returns a `VectorDBQueryResult` object, which may be the "class" you mentioned.

About the `# type: ignore`, it is only for `point.vector` that has a complicated structure (many combinations of `Union`). In our situation, some of these types will not appear here, so we use the notation to reduce the code complexity.